src/components/CodeBox.vue

```diff
@@ -1,30 +1,42 @@
 <script setup lang="ts">
-import { WordType, ICode } from '../models/models';
+import { WordType, ICode, ISolution, Solution, IBug, Bug, ErrorType } from '../models/models';
 import { CodeVisualizer } from '../models/code-visualizer';
 import { ref, watch } from 'vue';
 
 const props = defineProps<{
   code: ICode;
-  rightWordId: number;
-  wrongWordId: number;
+  feedbackSolution: Array<boolean>;
 }>();
 
 const emit = defineEmits<{
-  (e: 'submitSolution', selectedWordId: number): void;
+  (e: 'submitSolution', selectedBugs: ISolution): void;
 }>();
 
 const newLine = WordType.NEWLINE;
 const tab = WordType.TAB;
 
-let submitted = false;
+var submitted = ref(false);
+var selectedBugs = ref(Array<IBug>());
 
 var codeVisualizer = new CodeVisualizer(props.code);
 const codeLines = ref(codeVisualizer.getCodeLineWords());
 
-function clickedButton(id: number) {
-  if (!submitted) {
-    submitted = true;
-    emit('submitSolution', id);
+function submit() {
+  if (!submitted.value) {
+    submitted.value = true;
```

Comment:
I'm not quite sure if this once-only mechanism is needed.
Would it be better to use a mechanism such as
`multiple errors exist -> user selects not enough/ incorrect results -> message 'there are more bugs is shown' or 'incorrect solution is shown' -> user can try further until he finds the solution -> repeat -> success`?

Reply:
I don't know if this is a good solution because this could end in a brute-force behavior to get into the next level if the player has no further idea where the bug could be. In the actual game-play the player gets direct feedback what he did wrong because bugs he selected that are no bugs and bugs that are bugs he didn't selected are marked in red and correct selected bugs in green. I think this is better because the player won't stuck with one code snipped but gets feedback what was wrong and has the ability to analyze his mistake.